### PR TITLE
Go binding demo

### PR DESCRIPTION
**Pull request by ksyeo1010**
- add go binding and demo

### CODE REVIEW

File: demo/go/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+Replace `${ACCESS_KEY}` with yours obtained from Picovoice Console and `${AUDIO_PATH}` with a path to an audio file you
+wish to transcribe.
+
+### Microphone Demo
+
+You need a working microphone connected to your machine for this demo. Run the following in the terminal:
+
+```console
+go run filedemo/cheetah_file_demo.go -access_key "${ACCESS_KEY}"
```
Comment (laves) [inline demo/go/README.md:63]:
```suggestion
go run micdemo/cheetah_mic_demo.go -access_key "${ACCESS_KEY}"
```

File: binding/go/cheetah.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+	// Audio sample rate accepted by Picovoice.
+	SampleRate = nativeCheetah.nativeSampleRate()
+
+	// Cheetah version
+	Version = nativeCheetah.nativeVersion()
+)
+
+// Init function for Cheetah. Must be called before attempting process
+func (cheetah *Cheetah) Init() error {
```
Comment (laves) [inline binding/go/cheetah.go:109]: Could be worth it to have a default constructor (`NewCheetah`), similar to the way we do in Rhino. That way they don't have to set endpoint duration, we can just set it to a default.

**Review by laves — CHANGES_REQUESTED** [review]

File: demo/go/micdemo/cheetah_mic_demo.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+func printAudioDevices() {
+	if devices, err := pvrecorder.GetAudioDevices(); err != nil {
+		log.Fatalf("Error: %s.\n", err.Error())
+	} else {
+		for i, device := range devices {
+			log.Printf("index: %d, device name: %s\n", i, device)
+		}
+	}
+}
```
[the remark this answers is not in the document]
Comment (ksyeo1010) [reply demo/go/micdemo/cheetah_mic_demo.go:137]: incldues fixes for 64 bit, I added it since I had the pr open.

File: binding/go/cheetah.go [hunk 4]
```diff
@@ -105,6 +105,15 @@ var (
 	Version = nativeCheetah.nativeVersion()
 )
 
+// Returns a Cheetah struct with the given modelPath with default parameters
+func NewCheetah(accessKey string, modelPath string) Cheetah {
```
Comment (laves) [inline binding/go/cheetah.go:109]: I wonder if we should just even do just `AccessKey` for this constructor? It's meant to be the super-easy, batteries-included interface, so maybe we shouldn't confuse them with the empty string == default model.

File: binding/go/README.md [hunk 5]
```diff
@@ -43,10 +43,10 @@ AccessKey also verifies that your usage is within the limits of your account. Ev
 Create an instance of the engine and transcribe audio:
 
 ```go
-import . "github.com/Picovoice/leopard/binding/go"
+import . "github.com/Picovoice/cheetah/binding/go"
 
 cheetah = Cheetah{AccessKey: "${ACCESS_KEY}", EndpointDuration: 1.0}
```
Comment (laves) [inline binding/go/README.md:48]: maybe you should use `NewCheetah` here

File: binding/go/cheetah.go [hunk 6]
```diff
@@ -105,11 +105,11 @@ var (
 	Version = nativeCheetah.nativeVersion()
 )
 
-// Returns a Cheetah struct with the given modelPath with default parameters
-func NewCheetah(accessKey string, modelPath string) Cheetah {
+// Returns a Cheetah struct with defualt parameters
```
Comment (laves) [inline binding/go/cheetah.go:108]:
```suggestion
// Returns a Cheetah struct with default parameters
```

**Review by laves — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -111,6 +113,19 @@ Using Android Studio, open [demo/android/CheetahDemo](/demo/android/CheetahDemo)
 
 Replace `"${YOUR_ACCESS_KEY_HERE}"` in the file [MainActivity.java](/demo/android/cheetah-demo-app/src/main/java/ai/picovoice/cheetahdemo/MainActivity.java) with your `AccessKey`.
 
+### Go Demo
+
+The demo requires `cgo`, which on Windows may mean that you need to install a gcc compiler like [Mingw](http://mingw-w64.org/doku.php) to build it properly.
```
Comment (kenarsa) [inline README.md:118]: `Mingw` >> `MinGW`

File: binding/go/README.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
+# Cheetah Speech-to-Text Engine
+
+Made in Vancouver, Canada by [Picovoice](https://picovoice.ai)
+
+Cheetah is an on-device streaming speech-to-text engine. Cheetah is:
+
+- Private; All voice processing runs locally.
+- Accurate [[1]](https://github.com/Picovoice/speech-to-text-benchmark#results)
+- Compact and Computationally-Efficient [[2]](https://github.com/Picovoice/speech-to-text-benchmark#results)
```
Comment (kenarsa) [inline binding/go/README.md:9]: `#results` >>> `#rtf` ... I think

File: README.md [hunk 9]
```diff
@@ -268,6 +283,39 @@ try {
...
+
+for {
+  partialTranscript, isEndpoint, err = cheetah.Process(getNextFrameAudio())
+  if isEndpoint {
+    finalTranscript, err = cheetah.Flush()
+  }
+}
+```
+
+Replace `${ACCESS_KEY}` with yours obtained from Picovoice Console.
```
Comment (kenarsa) [inline README.md:317]: note about `release`

File: demo/go/README.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+Made in Vancouver, Canada by [Picovoice](https://picovoice.ai)
+
+## Cheetah
+
+Cheetah is an on-device streaming speech-to-text engine. Cheetah is:
+
+- Private; All voice processing runs locally.
+- Accurate [[1]](https://github.com/Picovoice/speech-to-text-benchmark#results)
+- Compact and Computationally-Efficient [[2]](https://github.com/Picovoice/speech-to-text-benchmark#results)
```
Comment (kenarsa) [inline demo/go/README.md:11]: `#rtf`

File: demo/go/README.md [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+Run the following in the terminal:
+
+```console
+go run filedemo/cheetah_file_demo.go \
+-input_audio_path "${AUDIO_PATH}" \
+-access_key "${ACCESS_KEY}"
+```
+
+Replace `${ACCESS_KEY}` with yours obtained from Picovoice Console and `${AUDIO_PATH}` with a path to an audio file you
```
Comment (kenarsa) [inline demo/go/README.md:55]: mention that we need WAV

File: demo/go/filedemo/cheetah_file_demo.go [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+		transcript += partial
+	}
+
+	final, err := c.Flush()
+	if err != nil {
+		log.Fatalf("Failed to flush: %v\n", err)
+	}
+	transcript += final
+
+	log.Println(transcript)
```
Comment (kenarsa) [inline demo/go/filedemo/cheetah_file_demo.go:105]: let's print incrementally.


let's release the cheetah

File: demo/go/micdemo/cheetah_mic_demo.go [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+			pcm, err := recorder.Read()
+			if err != nil {
+				log.Fatalf("Error: %v.\n", err)
+			}
+
+			partial, isEndpoint, err := c.Process(pcm)
+			if err != nil {
+				log.Fatalf("Process error: %v\n", err)
+			}
+			print(partial)
```
Comment (kenarsa) [inline demo/go/micdemo/cheetah_mic_demo.go:109]: do we need to flush?

File: demo/go/micdemo/cheetah_mic_demo.go [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+			if outputWav != nil {
+				for outputBufIndex := range pcm {
+					outputWav.WriteFrame(pcm[outputBufIndex])
+				}
+			}
+		}
+	}
+}
+
+func printAudioDevices() {
```
Comment (kenarsa) [inline demo/go/micdemo/cheetah_mic_demo.go:129]: lets release the cheetah at the end

**Review by kenarsa — CHANGES_REQUESTED** [review]

File: demo/go/filedemo/cheetah_file_demo.go (hunk 12 above)
Comment (ksyeo1010) [reply demo/go/filedemo/cheetah_file_demo.go]: release cheetah is done right after init with `defer cheetah.Delete()` <- how usually go does it

**Pull request merged**